src/interpreter/executor/interpret.js

```diff
@@ -0,0 +1,332 @@
+"use strict";
+// This is our 'state' when interpreting the code
+//  Use this enum as a flag for certain logic
+const stateEnum = {
+    DEFAULT : "default",
+    ACCEPTING_VAR : "accepting variable",
+    ASSIGNING_VAR : "assigning variable"
+}
+
+function Variable() {
+    this.name = "";
+    this.value = "";
+}
+
+// Our master state that we will output to visualizer
+var masterRep = [];
+
+// Single line representation we will push to master Rep
+var currentLineRep = [];
+
+// Current interpreter state for program
+var currentState = stateEnum.DEFAULT;
+
+// Current step in the user input code
+var currentStep = 0;
+
+// Current object we are adding data to
+var currentDataHold = null;
+
+//  Flag to tell code editor to be writable or not
+var isRunning = false;
+
+// Global var to bump out of interpreting if in a block comment
+var isBlockComment = false;
+
+//  List to hold user code, use a list so we can index on stepCount
+var userCode = []
+
+// Setup for when we initially check syntax
+function setup(inputCode) {
+    
+    //Lock  the code editor
+    isRunning = true;
+    
+    //Check validity of inputCode first
+    try {
+        new Function(inputCode);
+    } catch(err) {
+        exitInterpret();
+        return false;
+    }
+    
+    userCode = inputCode.split(/[\r\n]+/)
+    return true;
+}
+
+// Cleanup function to be called whenever exiting interpreter
+function exitInterpret() {
+    
+    masterRep = [];
+    currentLineRep = [];
+    currentState = stateEnum.DEFAULT;
+    currentStep = 0;
+    isRunning = false;
+    userCode = [];
+    isBlockComment = false;
+    currentDataHold = null;
+}
+
+function interpretLine(lineNumber) {
+
+    //Handling multi statements on one line
+    var statements = userCode[lineNumber].split(';');
+    
+    for (var i = 0; i < statements.length; i++) {
```

Comment:
NIT: When you simply need to iterate over an array list in its entirety, try using the forEach function for arrays for readability.

http://thejsguy.com/2016/07/30/javascript-for-loop-vs-array-foreach.html

Reply:
I can change the outer loop to a for each however, the continue would have to be replaced with a function return.
The inner loop can also be replaced however, the continues would once again shift to returns and in order to add the break functionality I would have to throw an exception.  

Do you think those changes would negate the readability of the for each header?

Reply:
Would this be a good compromise? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

I think this mitigates both problems.